kind/pkg/cluster/config/errors.go

```diff
@@ -0,0 +1,48 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import "bytes"
+
+// Errors implements error and contains all config errors
+// This is returned by Config.Validate
+type Errors struct {
+	errors []error
+}
```

Comment:
yeah it's very similar, but I wanted control over the error formatting initially. I also wanted to avoid pulling in any of k/k initially, but it could be worth switching... hmm

Reply:
I also expect callers checking the type to be able to detect this type and call `.Errors()` to better debug the errors encountered, so if we did switch to embedding Aggregate we'd need to expose that still.